### ISSUE TITLE
Move Flow/Neos specific classes to Neos namespace

### ISSUE THREAD

**Issue by Sebobo** [TYPO3/Surf#49]
- [x] Move classes to new namespace
- [x] Adapt cli commands to work with older Flow versions and new ones

**Comment by Sebobo** [comment]
With the release of Neos 3 and Flow 4 this change is getting bigger.

**Comment by p34ch** [comment]
Are there any plans yet when the new version of Surf shall be released to support deployment of Flow 4.0 applications?

How is deployment of these updated (Flow 4.0) applications possible in the meantime?

**Comment by Sebobo** [comment]
@p34ch I'm a bit busy right now to finalize the change but I hope to get it done soon.
But of course help is welcome!

For now you just need to replace two tasks to deploy Flow 4.0 right now:

```
// Surf doesn't support the new namespace yet so we have to do it by hand for now
$workflow
    ->removeTask('TYPO3\\Surf\\Task\\TYPO3\\Flow\\MigrateTask')
    ->removeTask('TYPO3\\Surf\\Task\\TYPO3\\Flow\\PublishResourcesTask');

$workflow->defineTask(
    'Xyz:Migrate',
    'TYPO3\\Surf\\Task\\TYPO3\\Flow\\RunCommandTask',
    [
        'command' => 'doctrine:migrate'
    ]
)->forStage('migrate', 'Xyz:Migrate');

$workflow->defineTask(
    'Xyz:PublishResources',
    'TYPO3\\Surf\\Task\\TYPO3\\Flow\\RunCommandTask',
    [
        'command' => 'resource:publish'
    ]
)->forStage('finalize', 'Xyz:PublishResources');
```

**Comment by p34ch** [comment]
@Sebobo Thanks very much for your reply!

I'm still getting an error, after I added your snippet to my deployment script configuration:

`Stage migrate`
`remotehost (Neos Flow) Xyz:Migrate`
`Warning: require(/home/httpd/vhosts/[...]/httpdocs/releases/20170227120113/Packages/Framework/TYPO3.Flow/Scripts/flow.php): failed to open stream: No such file or directory in /home/httpd/vhosts/[...]/httpdocs/releases/20170227120113/flow on line 18  Warning: require(/home/httpd/vhosts/[...]/httpdocs/releases/20170227120113/Packages/Framework/TYPO3.Flow/Scripts/flow.php): failed to open stream: No such file or directory in /home/httpd/vhosts/[...]/httpdocs/releases/20170227120113/flow on line 18  Fatal error: require(): Failed opening required '/home/httpd/vhosts/[...]/httpdocs/releases/20170227120113/Packages/Framework/TYPO3.Flow/Scripts/flow.php' (include_path='.:/usr/share/pear:/usr/share/php') in /home/httpd/vhosts/[...]/httpdocs/releases/20170227120113/flow on line 18
`

At first a create a new SimpleWorkflow. Afterwards I add your snipped, then I link this workflow to my deployment. Next I create a node, add this node to a new application and assign this application to the deployment as well.

Do you have any hints where/how I might fix this issue for deploying my Flow 4.0 application with surf?

**Comment by Sebobo** [comment]
@p34ch you should one time delete your surf cache on the target system.
`composer install` doesn't update the flow binary if you reuse the checked out project and don't recreate it for every build.
That should help I guess.
